File: spring-web/src/main/java/com/example/springwebex/model/pagenation/ResponseHeader.java
package com.example.springwebex.model.pagenation;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

import lombok.Data;

@Data
@JsonInclude(Include.NON_NULL)  // serialize
@JsonIgnoreProperties(ignoreUnknown = true) // deserialize
public class ResponseHeader {
    Boolean isSuccessful;
    Integer resultCode;
    String resultMessage;
}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/model/pagenation/ResponseHeader.kt
package com.example.springwebex.model.pagenation

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
data class ResponseHeader(
    var isSuccessful: Boolean? = null,
    var resultCode: Int? = null,
    var resultMessage: String? = null
)
